Retail Transaction Dataset
Dataset : https://www.kaggle.com/datasets/prasad22/retail-transactions-dataset
User : Fairuz30

In [26]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [27]:
df = pd.read_csv("Masukkan Destinasi File CSV Disini")

In [28]:
df.head()

Transaction_ID                 Date     Customer_Name  \
0      1000000000  2020-12-21 19:42:52   Cheyenne Newman   
1      1000000001  2020-07-06 07:45:16  Emily Fitzgerald   
2      1000000002  2021-10-02 06:28:44      Michael Webb   
3      1000000003  2022-01-10 05:39:02      Kimberly Lin   
4      1000000004  2021-10-13 07:28:47   Cathy Hernandez   

                                             Product  Total_Items  Total_Cost  \
0                                       ['Hair Gel']            6       12.77   
1         ['Tuna', 'Bread', 'Tissues', 'Trash Bags']            5       13.88   
2                         ['Jam', 'Soap', 'Ketchup']            7       47.02   
3                                      ['BBQ Sauce']            9       83.86   
4  ['Hand Sanitizer', 'Bread', 'Extension Cords',...            4       30.55   

   Payment_Method      City         Store_Type  Discount_Applied  \
0      Debit Card  New York  Convenience Store              True   
1      Debit Card   Houston        Supermarket             False   
2      Debit Card     Miami  Convenience Store             False   
3  Mobile Payment   Seattle     Warehouse Club              True   
4      Debit Card   Houston     Warehouse Club             False   

  Customer_Category  Season                   Promotion  
0           Student  Winter                         NaN  
1      Professional    Fall      BOGO (Buy One Get One)  
2       Young Adult  Winter                         NaN  
3    Senior Citizen  Summer  Discount on Selected Items  
4    Senior Citizen  Spring                         NaN

In [29]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 30000 entries, 0 to 29999
Data columns (total 13 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Transaction_ID     30000 non-null  int64  
 1   Date               30000 non-null  object 
 2   Customer_Name      30000 non-null  object 
 3   Product            30000 non-null  object 
 4   Total_Items        30000 non-null  int64  
 5   Total_Cost         30000 non-null  float64
 6   Payment_Method     30000 non-null  object 
 7   City               30000 non-null  object 
 8   Store_Type         30000 non-null  object 
 9   Discount_Applied   30000 non-null  bool   
 10  Customer_Category  30000 non-null  object 
 11  Season             30000 non-null  object 
 12  Promotion          20009 non-null  object 
dtypes: bool(1), float64(1), int64(2), object(9)
memory usage: 2.8+ MB


In [30]:
df.duplicated().sum()

0

In [31]:
df['Product'].value_counts()

Product
['Toothpaste']                                                                   121
['Yogurt']                                                                        94
['Milk']                                                                          92
['Tuna']                                                                          90
['Soda']                                                                          88
                                                                                ... 
['Mustard', 'Light Bulbs', 'Laundry Detergent', 'Bath Towels', 'Trash Bags']       1
['Cheese', 'Shrimp', 'Milk', 'Rice', 'Paper Towels']                               1
['Hand Sanitizer', 'Diapers']                                                      1
['Bath Towels', 'Cleaning Spray', 'Pancake Mix', 'Plant Fertilizer', 'Apple']      1
['Hair Gel', 'Mop', 'Peanut Butter']                                               1
Name: count, Length: 21967, dtype: int64

berasarkan hasil eksekusi berikut dapat diketahui bahwa Toothpaste atau pasta gigi merupakan komoditas yang paling banyak terjual

In [32]:
df_counts =df.groupby('Customer_Name')['Product'].value_counts()
print(df_counts)

Customer_Name    Product                                                          
Aaron Allen      ['Pickles', 'Deodorant', 'Tea']                                      1
Aaron Andrews    ['Water', 'Chips', 'Extension Cords', 'Dustpan', 'Shrimp']           1
Aaron Armstrong  ['Pickles', 'Power Strips']                                          1
Aaron Arnold     ['Butter', 'Mayonnaise']                                             1
Aaron Brennan    ['Spinach', 'Dishware', 'Broom']                                     1
                                                                                     ..
Zoe Chaney       ['Olive Oil']                                                        1
Zoe Dorsey       ['Spinach']                                                          1
Zoe Jones        ['Toothpaste', 'Tomatoes']                                           1
Zoe Larson       ['Mustard', 'Broom']                                                 1
Zoe Mendoza      ['Toothpaste', 'Carr

Mengetahui ada berapa banyak jenis pelanggan didalam dataset

In [33]:
df['Customer_Category'].value_counts()

Customer_Category
Student           3904
Homemaker         3798
Teenager          3780
Middle-Aged       3747
Retiree           3745
Senior Citizen    3702
Young Adult       3686
Professional      3638
Name: count, dtype: int64

In [34]:
df_counts = df.groupby('Customer_Category')['Store_Type'].value_counts()
print(df_counts)

Customer_Category  Store_Type       
Homemaker          Pharmacy             658
                   Department Store     646
                   Convenience Store    635
                   Supermarket          631
                   Warehouse Club       624
                   Specialty Store      604
Middle-Aged        Supermarket          654
                   Warehouse Club       640
                   Specialty Store      630
                   Department Store     615
                   Pharmacy             612
                   Convenience Store    596
Professional       Supermarket          621
                   Pharmacy             615
                   Department Store     614
                   Specialty Store      597
                   Convenience Store    597
                   Warehouse Club       594
Retiree            Warehouse Club       655
                   Pharmacy             636
                   Specialty Store      634
                   Supermarket         

dapat diketahui bahwa :
    Homemaker atau biasa disebut orang yang banyak menghabiskan waktu dirumah sering mengunjungi pharmacy atau klinik (ditunjukkan sebanyak 658 orang)

    Middle-aged atau orang berumur 40 - 50 tahun sering mengunjungi supermarket (654 0rang)

    Professional atau seorang yang ahli sering mengunjungi supermarket (ditunjukkan sebanyak 621 orang)

    Retiree atau pensiunan sering mengunjungi warehouse club (ditunjukkan sebanyak 655)

    dan seterusnya

berdasarkan hal diatas dapat diketahui bahwa setiap golongan orang memiliki tujuan tempat pembelian yang berbeda beda, dan ini bisa dijadikan salah satu faktor oleh pihak tempat penjualan dalam menentukan diskon untuk tipe pembeli tersebut.

# Diskon

In [35]:
customer1=df.groupby('Customer_Category')['Discount_Applied'].value_counts()
print(customer1)

Customer_Category  Discount_Applied
Homemaker          False               1902
                   True                1896
Middle-Aged        False               1908
                   True                1839
Professional       False               1826
                   True                1812
Retiree            False               1889
                   True                1856
Senior Citizen     False               1860
                   True                1842
Student            False               1976
                   True                1928
Teenager           True                1909
                   False               1871
Young Adult        True                1872
                   False               1814
Name: count, dtype: int64


berdasarkan eksekusi kode diatas dapat dijelaskan bahwa terdapat kategori pelanggan yang mendapat diskon dan yang tidak.

In [36]:
discount_type=df.groupby('Discount_Applied')['Promotion'].value_counts()
print(discount_type)

Discount_Applied  Promotion                 
False             BOGO (Buy One Get One)        5044
                  Discount on Selected Items    4992
True              BOGO (Buy One Get One)        5024
                  Discount on Selected Items    4949
Name: count, dtype: int64


dapat diketahui pada dataset diatas terdapat dua jenis diskon yaitu BOGO dan Diskon untuk barang tertentu

In [37]:
Customer_discount=df.groupby('Customer_Category')['Promotion'].value_counts()
print(Customer_discount)

Customer_Category  Promotion                 
Homemaker          Discount on Selected Items    1294
                   BOGO (Buy One Get One)        1250
Middle-Aged        BOGO (Buy One Get One)        1298
                   Discount on Selected Items    1210
Professional       Discount on Selected Items    1208
                   BOGO (Buy One Get One)        1197
Retiree            BOGO (Buy One Get One)        1272
                   Discount on Selected Items    1271
Senior Citizen     BOGO (Buy One Get One)        1250
                   Discount on Selected Items    1198
Student            BOGO (Buy One Get One)        1304
                   Discount on Selected Items    1283
Teenager           BOGO (Buy One Get One)        1259
                   Discount on Selected Items    1249
Young Adult        BOGO (Buy One Get One)        1238
                   Discount on Selected Items    1228
Name: count, dtype: int64


dan hasil eksekusi kode diatas adalah persebaran diskon di antara masing masing tipe pelanggan yang ada di dataset.

#Jenis Pembayaran

In [38]:
Customer_destination=df.groupby('Customer_Category')['Payment_Method'].value_counts()
print(Customer_destination)

Customer_Category  Payment_Method
Homemaker          Cash              979
                   Credit Card       944
                   Debit Card        941
                   Mobile Payment    934
Middle-Aged        Cash              966
                   Credit Card       952
                   Mobile Payment    917
                   Debit Card        912
Professional       Debit Card        921
                   Mobile Payment    912
                   Cash              906
                   Credit Card       899
Retiree            Credit Card       968
                   Debit Card        932
                   Mobile Payment    930
                   Cash              915
Senior Citizen     Debit Card        990
                   Mobile Payment    917
                   Cash              907
                   Credit Card       888
Student            Debit Card        991
                   Credit Card       984
                   Mobile Payment    978
                   Cash

eksekusi kode diatas dapat diketahui bahwa terdapat 4 jenis pembayaran yaitu : 

1. Debit Card 
2. Credit Card
3. Cash
4. Mobile Payment

#mengetahui waktu pelanggan paling banyak mendatangi ke toko pada season apa?

In [47]:
Season_Customer=df.groupby('Customer_Category')['Season'].value_counts()
print(Season_Customer)

Customer_Category  Season
Homemaker          Winter     965
                   Summer     963
                   Spring     954
                   Fall       916
Middle-Aged        Summer     966
                   Spring     959
                   Fall       913
                   Winter     909
Professional       Spring     935
                   Fall       928
                   Summer     912
                   Winter     863
Retiree            Fall       975
                   Spring     932
                   Winter     931
                   Summer     907
Senior Citizen     Spring     958
                   Winter     933
                   Summer     909
                   Fall       902
Student            Summer    1021
                   Fall       984
                   Spring     967
                   Winter     932
Teenager           Summer     960
                   Fall       946
                   Winter     940
                   Spring     934
Young Adult        Fal

Dapat diketahui bahwa terdapat 4 season yang ada dalam dataset, dan menurut summer merupakan waktu yang paling banyak dipilih oleh beberapa jenis tipe pelanggan untuk membeli barang di tempat penjualan.